graspologic/pipeline/embed/adjacency_spectral_embedding.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Optional, Union
+import warnings
+
+import networkx as nx
+import numpy as np
+
+from . import __SVD_SOLVER_TYPES  # from the module init
+from ._elbow import _index_of_elbow
+from .embeddings import Embeddings
+from ..preconditions import (
+    check_argument,
+    check_argument_types,
+    check_optional_argument_types,
+    is_real_weighted,
+)
+from graspologic.embed import AdjacencySpectralEmbed
+from graspologic.utils import is_fully_connected, pass_to_ranks
+
+
+def adjacency_spectral_embedding(
+    graph: Union[nx.Graph, nx.DiGraph],
+    dimensions: int = 100,
+    elbow_cut: Optional[int] = None,
+    svd_solver_algorithm: str = "randomized",
+    svd_solver_iterations: int = 5,
+    svd_seed: Optional[int] = None,
+    weight_attribute: str = "weight",
+) -> Embeddings:
+    """
+    Given a weighted directed or undirected networkx graph (*not* multigraph),
```

Comment:
doesn't have to be weighted

Reply:
mathematically at least - you're saying it does for this function?

Reply:
I was, but we actually allow unweighted and we make them into weighted, so it's a good catch